### PR TITLE
Issue-17 deletion protection

### LINKED ISSUE #17: Making delete protection for RDS configurable
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* ***If you have submitted a pull request addressing the same issue, please leave a comment!***

***

### Description

In some environments, RDS database should never be deleted. Allow this parameter to be configurable https://www.terraform.io/docs/providers/aws/r/db_instance.html#deletion_protection thus will avoid any accidental deletion of RDS cluster.

### New or Affected Resource(s)

* aws_db_instance

### Potential Terraform Configuration Changes

```hcl
# Copy-paste your Terraform configurations here.
# For large Terraform configs, please use a service like Google Drive and share a link to the ZIP file.
```

### References

* #0000

### PR DESCRIPTION
**Pull request by alexandersperling**
* add new variable for deletion protection
* add parameter to main config file

### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" comments, they generate extra noise for pull request followers and do not help prioritize the request

***

Fixes #17 

***

Release note for [CHANGELOG](https://github.com/traveloka/terraform-aws-rds-postgres/blob/master/CHANGELOG.md):

```release-note
NOTES:

* deletion protection for db instance is now configurable, default value is false 

ENHANCEMENTS:

* deletion protection for db instance is now configurable, default value is false
```

***

Output from `terraform plan` command from changes you propose.

```

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + aws_db_instance.this
      id:                                    <computed>
      address:                               <computed>
      allocated_storage:                     "20"
      allow_major_version_upgrade:           "false"
      apply_immediately:                     "false"
      arn:                                   <computed>
      auto_minor_version_upgrade:            "false"
      availability_zone:                     <computed>
      backup_retention_period:               "7"
      backup_window:                         <computed>
      ca_cert_identifier:                    <computed>
      character_set_name:                    <computed>
      copy_tags_to_snapshot:                 "true"
      db_subnet_group_name:                  <computed>
      deletion_protection:                   "false"
      endpoint:                              <computed>
      engine:                                "postgres"
      engine_version:                        <computed>
      final_snapshot_identifier:             "${local.final_snapshot_identifier}"
      hosted_zone_id:                        <computed>
      identifier:                            "${random_id.db_identifier.hex}"
      identifier_prefix:                     <computed>
      instance_class:                        "db.t2.micro"
      iops:                                  "0"
      kms_key_id:                            <computed>
      license_model:                         <computed>
      maintenance_window:                    "mon:00:00-mon:03:00"
      monitoring_interval:                   "60"
      monitoring_role_arn:                   "arn:aws:iam::123456789012:role/S3Access"
      multi_az:                              "true"
      name:                                  <computed>
      option_group_name:                     <computed>
      parameter_group_name:                  "groupName"
      password:                              <sensitive>
      performance_insights_enabled:          "false"
      performance_insights_kms_key_id:       <computed>
      performance_insights_retention_period: <computed>
      port:                                  "5432"
      publicly_accessible:                   "false"
      replicas.#:                            <computed>
      resource_id:                           <computed>
      skip_final_snapshot:                   "false"
      status:                                <computed>
      storage_encrypted:                     "true"
      storage_type:                          "gp2"
      tags.%:                                <computed>
      timezone:                              <computed>
      username:                              "postgres"
      vpc_security_group_ids.#:              "1"
      vpc_security_group_ids.809200604:      "sg-ca3d48a2"

  + random_id.db_identifier
      id:                                    <computed>
      b64:                                   <computed>
      b64_std:                               <computed>
      b64_url:                               <computed>
      byte_length:                           "8"
      dec:                                   <computed>
      hex:                                   <computed>
      prefix:                                "service-postgres-"

  + random_id.password
      id:                                    <computed>
      b64:                                   <computed>
      b64_std:                               <computed>
      b64_url:                               <computed>
      byte_length:                           "8"
      dec:                                   <computed>
      hex:                                   <computed>


Plan: 3 to add, 0 to change, 0 to destroy.


```

### CODE REVIEW

**Comment by sshvetsov** [comment]
Hi @alexandersperling please fix the PR description content to follow PR template provided.

**Comment by alexandersperling** [comment]
Hi @sshvetsov looks like I missed the `terraform plan` part. Updated it now.

**Comment by rafikurnia** [comment]
Halo @alexandersperling again thanks for the PR..
Related the build you can ignore it, that's known issue on our side.
I will get back to you to review this PR mid next week

File: variables.tf [hunk 1]
```diff
@@ -172,3 +172,8 @@ variable "monitoring_role_arn" {
   type        = "string"
   description = "The ARN for the IAM role that permits RDS to send enhanced monitoring metrics to CloudWatch Logs"
 }
+
+variable "deletion_protection" {
+  type    = "string"
+  default = "false"
```
Comment (sumodirjo) [inline variables.tf]: Hi @alexandersperling ,
Please add description for this variable. Thank you

**Comment by sumodirjo** [comment]
This feature already added in #23

**Pull request closed without merging**